### PR TITLE
Windows Installer

### PR DESCRIPTION
**Pull request by rprouse**
**Ready to merge**

See #419 

I have split the installer into two installers, one **NUnit.Tools** and the second **NUnit.Framework**. 

Tools always installs the latest version and currently includes the engine and the console runner.

Framework installs side by side into a versioned subdirectory. When both installer are installed, it currently looks like this.

```
Program Files (x86)
    NUnit.org
        tools
            engine and console files (1)
        framework
            v3.0.xxx.xxx
                net-2.0
                    nunit.framework.dll
                    nunit.framework.xml
                    nunitlite.dll
                net-3.5 etc...
```

(1) The engine and console are currently in the same directory even though they are two features in the installer. I will move these into engine and console directories once the engine can register itself and be found elsewhere.

## Todo ##

- ~~Add an icon for the install~~
- ~~Brand the install dialogs with icon and graphics~~
- ~~Add the assemblies to the registry so they can be referenced.~~
- ~~Add msbuild targets to nunit.proj~~

### CODE REVIEW

**Comment by rprouse** [comment]
I've changed the assembly titles for the framework and the runners to include the target framework. They are now distinguishable in the add references dialog.

2015-01-03 17_03_19-reference manager - nunit-v3

**Comment by rprouse** [comment]
I've added `PackageAllInstalls`, `PackageFrameworkInstall` and `PackageRunnersInstall` targets to NUnit.proj. We can now build installs with msbuild on the command line.

**Comment by rprouse** [comment]
This is ready to merge. It isn't 100% complete, but I would prefer to make additional changes with separate issues and smaller pull requests. I want to get this out so that people can try it.

This currently creates two installers either in the Visual Studio build, or through the command line MSBuild. Changing them to two merge modules and then three installers (framework, runners and everything) can be a separate issue.

This requires that developers have Wix installed. If they don't, like Silverlight, it will give you a warning the first time you open the solution and the projects will be unloaded.

The installers currently do not register the engine since we haven't decided yet how that will work.

File: install/runners/nunit-tools.wxs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        <!-- ***************************************************************** -->
+        <!-- **********      Define the NUnit feature tree          ********** -->
+        <!-- ***************************************************************** -->
+        
+        <Feature Id="NUNIT.RUNNERS" 
+                 ConfigurableDirectory="INSTALLDIR" 
+                 Level="1" 
+                 Title="NUnit Runners" 
+                 Display="expand" 
+                 Description="Installs the NUnit engine and tools that allow you to run tests.">
```
Comment (CharliePoole) [inline install/runners/nunit-tools.wxs]: "and console runner" ... there are no other tools yet.

File: install/runners/nunit-tools.wxs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+    <!-- ***************************************************************** -->
+    <!-- **********       Define Directory Structure            ********** -->
+    <!-- ***************************************************************** -->
+    
+	<Fragment>
+		<Directory Id="TARGETDIR" Name="SourceDir">
+			<Directory Id="ProgramFilesFolder">
+                <Directory Id="INSTALLDIR" Name="NUnit.org">
+                    <Directory Id="TOOLS" Name="tools" />
```
Comment (CharliePoole) [inline install/runners/nunit-tools.wxs]: console?

File: install/runners/nunit-tools.wxs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+        <ComponentGroup Id="NUNIT.CONSOLE" Directory="TOOLS">
+            <Component Id="NUNIT_AGENT">
+                <File Id="nunit_agent.exe" 
+                      ProcessorArchitecture="msil" 
+                      Source="$(var.nunit-agent.TargetPath)" />
+                <File Id="nunit_agent.exe.config" 
+                      Source="$(var.nunit-agent.TargetPath).config" />
+            </Component>
+            <Component Id="NUNIT_AGENT_X86">
```
Comment (CharliePoole) [inline install/runners/nunit-tools.wxs]: Agent isn't part of the console, it's part of the engine. When we have other runners, they will use the same engine components, including agent, as the console.

**Comment by CharliePoole** [comment]
The only thing that really needs changing in order to merge is to move the agent files into the engine component group and feature. Otherwise, installing only the engine will give you an incomplete install, in fact one that won't work using the defaults that run tests in another process.

We can deal with directory naming, etc. when you do the next stage.

File: install/runners/nunit-tools.wxs (hunk 2 above)
Comment (rprouse) [inline install/runners/nunit-tools.wxs]: I've made the changes based on your other comments, but left this one. I don't think engine, console or tools is right for this directory name. I am hoping it becomes clearer once we get the engine loading issue sorted out. If everything is together, maybe bin for executables and lib for the framework?

**Comment by CharliePoole** [comment]
Here's what I still see in nunit-tools.wxs...

Engine Feature
   Engine ComponentGroup
      Engine Component
      Engine.Api Component
      Agent_X86 Component
      Console Component
Console Feature
   Console ComponentGroup
      Agent Component

We need the Console Component in the Console ComponentGroup and the other four in the Engine ComponentGroup.

**Comment by rprouse** [comment]
I shouldn't rush fixes, sorry about that. It is fixed now.

**Pull request merged**